src/meta/meta_http_service.cpp

```diff
@@ -769,6 +769,39 @@ void meta_http_service::start_compaction_handler(const http_request &req, http_r
     update_app_env(info.app_name, keys, values, resp);
 }
 
+void meta_http_service::update_scenario_handler(const http_request &req, http_response &resp)
+{
+    if (!redirect_if_not_primary(req, resp)) {
+        return;
+    }
+
+    // validate paramters
+    usage_scenario_info info;
+    bool ret = json::json_forwarder<usage_scenario_info>::decode(req.body, info);
+    if (!ret) {
+        resp.body = "invalid request structure";
+        resp.status_code = http_status_code::bad_request;
+        return;
+    }
+    if (info.app_name.empty()) {
+        resp.body = "app_name should not be empty";
+        resp.status_code = http_status_code::bad_request;
+        return;
+    }
```

Comment:
There are many similar code like this, maybe we can use macros to simplify code in future refactor.